css/amp.css

```diff
@@ -153,6 +153,14 @@ html.i-amphtml-ios-embed.i-amphtml-ios-overscroll > #i-amphtml-wrapper {
   position: relative;
 }
 
+/**
+ * Children of responsive elements should not be able to influence the height
+ * which they do when they are statically positioned.
+ */
+.i-amphtml-layout-responsive > :not(i-amphtml-sizer) {
```

Comment:
The problem is that we don't even get to stubbing phase at all. The #10294 this well, imho. But one partial option could be to also do something like this:
```
[layout=responsive]:unresolved > * {
  display: none;
}
```

Or, as the matter of fact, we could even do:
```
:unresolved > * {
  display: none;
}
```

With two caveats:
 1. `:unresolved` is not available everywhere yet, not even in Safari. Not sure if it will ever be supported. So, we may need to instead do a `[layout=X]:not(.i-amphtml-element)` dance. Which would be riddled with it's own set of issues, including the fact that we don't always specify `layout` - sometimes is defaulted, e.g. based on `width="auto"` attribute.
 2. Instead of estimating measures up - we will tend to estimate them down with this solution - i.e. we'll be loading more elements even though they'd not be loaded with the correct size. In other words, sometimes now, an 100px-height element is measured to be 1000px, but now it will be measured as 0px - pick your poison.